envs/pybullet_env.py

```diff
@@ -77,17 +87,22 @@ def step(self, action):
     if motion_primative == self.PICK_PRIMATIVE:
       self.ur5.pick(pos, rot, self.pick_offset, dynamic=self.dynamic)
     elif motion_primative == self.PLACE_PRIMATIVE:
-      self.ur5.place(pos, rot, self.place_offset, dynamic=self.dynamic)
+      if self.ur5.is_holding:
+        self.ur5.place(pos, rot, self.place_offset, dynamic=self.dynamic)
     elif motion_primative == self.PUSH_PRIMATIVE:
       pass
     else:
       raise ValueError('Bad motion primative supplied for action.')
 
     if self.ur5.is_holding:
       self.ur5.moveTo(self.rest_pose[0], self.rest_pose[1], dynamic=True)
+      self.ur5.checkGripperClosed()
     else:
       self.ur5.moveTo(self.rest_pose[0], self.rest_pose[1], dynamic=self.dynamic)
 
+    for _ in range(100):
```

Comment:
What is the purpose of this loop?

Reply:
After transition there might be a drop. The loop is for waiting for a stable status. Without it the block might be dropped and float in the air

Reply:
Do you think this would make sense to move to the sub-classes? I could see different envs having to wait a different amount of time between actions and it should be easy to do.